langkit/regexes.py

```diff
@@ -4,44 +4,70 @@
 )
 import json
 import re
-import pkg_resources
 from typing import Optional
 from logging import getLogger
+from . import LangKitConfig
 
 logger = getLogger(__name__)
 
-# If not stated otherwise, suggested patterns in the default file should reflect US patterns.
-pattern_json_filename = "pattern_groups.json"
-json_path = pkg_resources.resource_filename(__name__, pattern_json_filename)
-
-try:
-    skip = False
-    with open(json_path, "r") as myfile:
-        _REGEX_GROUPS = json.load(myfile)
-    regex_groups = []
-    for group in _REGEX_GROUPS:
-        compiled_expressions = []
-        for expression in group["expressions"]:
-            compiled_expressions.append(re.compile(expression))
-        regex_groups.append(
-            {"name": group["name"], "expressions": compiled_expressions}
-        )
-except FileNotFoundError:
-    skip = True
-    logger.warning(f"Could not find {pattern_json_filename}")
-except json.decoder.JSONDecodeError:
-    skip = True
-    logger.warning(f"Could not parse {pattern_json_filename}")
-
-if not skip:
+lang_config = LangKitConfig()
+
+
+class PatternLoader:
+    def __init__(self):
+        self.config: LangKitConfig = LangKitConfig()
+        self.regex_groups = self.load_patterns()
+
+    def load_patterns(self):
+        json_path = self.config.pattern_file_path
+        try:
+            skip = False
+            with open(json_path, "r") as myfile:
+                _REGEX_GROUPS = json.load(myfile)
+            regex_groups = []
+            for group in _REGEX_GROUPS:
+                compiled_expressions = []
+                for expression in group["expressions"]:
+                    compiled_expressions.append(re.compile(expression))
+                regex_groups.append(
+                    {"name": group["name"], "expressions": compiled_expressions}
+                )
+        except FileNotFoundError:
+            skip = True
+            logger.warning(f"Could not find {json_path}")
+        except json.decoder.JSONDecodeError:
+            skip = True
+            logger.warning(f"Could not parse {json_path}")
+        if not skip:
+            return regex_groups
+        return None
+
+    def set_config(self, config: LangKitConfig):
+        self.config = config
+
+    def update_patterns(self):
+        self.regex_groups = self.load_patterns()
+
+    def get_regex_groups(self):
+        return self.regex_groups
+
+
+pattern_loader = PatternLoader()
```

Comment:
I think we need to follow up with an init hook for some of these global instances.